---
title: About
---

<style>
/**/
.introduction{display:inline-block; /*padding-right:10px;*/ padding-bottom:50px;}
.profile_photo_text{float:left; width: 35%;}
.introduction_text{padding-left:25px;float:right;width:65%}
.download_link{padding-left:25px;float:right;width:65%}
.photo_text{float:left; width:90%; font-size: 12px; }
.link_text{float:left; width:90%; font-size: 14px; font-weight:bold;}
.introduction_text p {font-size: 14px;width:100%;}
/**/
/**/
.box{display: flex;vertical-align:top;}
.title_year{width:10%;font-weight: bold; /*padding-left:30px;*/ color: black;}
.Publications_text{font-size: 14px; padding-left:30px; width:90%;float:right;}
/**/
.footer{text-align: center;
		margin: 60px;
		}
</style>

<div class="introduction">
	<div class="profile_photo_text">
		<div class="images">
		<img src="./assets/images/lai_headshot.jpeg" width="314" height="392" />
		</div>
		<div class="photo_text">
			<p>Director of the Cornell Center for Vertebrate Genomics.
				</br>
				Director of the Cornell EpiGenomics Core Facility.
				</br>
				Assistant Research Professor in Molecular Biology and Genetics.
				</br>
				Assistant Research Professor in Computational Biology.
				</br>
				458 Biotechnology Building, Cornell, Ithaca, NY, USA 14850</p>
		</div>
	</div>
	<div class="introduction_text">
			<h1>William KM Lai</h1>
			<P>The primary focus of my research program is to understand how protein-complexes target the genome, how protein binding changes across cell states, and how mis-regulation of these interactions can result in disease-phenotypes. My group combines novel genomic assays with custom bioinformatic software development. We pursue ‘big-data’ projects that generate thousands of unique (epi)genomic datasets across human, mouse, and yeast model systems.
                                                     </br>                                       </br>
     As the Director of the Cornell EpiGenomics Core facility, my group works hand-in-hand with physician scientists at Cornell Weill medical campus to apply our epigenomic technology and algorithmic approaches to biomedical specimens to investigate the fundamental nature of human diseases. Combining wet-bench biochemical genomic technology with analytical bioinformatic algorithmic approaches allows us to understand the fundamental rules of protein-DNA interactions and how these rules are broken in diseased patient samples.
																									 	 </br>                                       </br>
     The volume and dimensionality of our data typically requires application of machine learning approaches as well as eXplainable AI algorithms. To support our analysis, we heavily leverage NSF-provided ACCESS resources. Our students are trained to perform advanced informatic analysis across a wide spectrum of heterogeneous and high-performance compute systems.
	 																									 </br>                                       </br>
		 The growing complexity of STEM research now frequently demands expertise in more domains of science than what any single research group is able to encompass. As the Director of the Cornell Center for Vertebrate Genomics, my mandate is to identify, develop, and support cross-discipline collaborations and foster an engaged community across the University. In a world that increasingly requires collaboration to leverage the latest and most powerful technologies, it is my privilege to support the researchers in the Center.
	 		</P>
			<p>The greatest biological discoveries are achieved when we can pair multiple distinct approaches (e.g., biochemistry, molecular biology, bioinformatics, etc.) to build something greater than the sum of its parts.</p>
	</div>
	<div class="download_link">
		<a href="./assets/cv/Lai_CV.pdf" download>Download CV</a>
	</div>
</div>

<div class="Publications">
	<h2>Publications</h2>
	<a href="/publications">View More</a>

	<div class="box">
	<div class="title_year">
		<p>2025</p>
	</div>
			<div class="Publications_text">
				<ul>
					<li><p><b><span style="color: black">Protein structure prediction and design for high-throughput computing.</span></b><br />
					<span style="color:"><u>Mathew VS,</u> <u>Kellogg GD,</u> <strong><u>Lai WKM.</u></strong> </span><br />
					<span style="color:black"><b>bioArxiv</b></span> 2025, <a href="https://doi.org/10.1101/2025.07.18.665594">https://doi.org/10.1101/2025.07.18.665594</a> </p>

					<li><p><b><span style="color: black">Adversarial attack of sequence-free enhancer prediction identifies chromatin architecture.</span></b><br />
					<span style="color:"><u>Gafur J,</u> <u>Lang OW,</u> <strong><u>Lai WKM.</u></strong> </span><br />
					<span style="color:black"><b>Bioinformatics</b></span> 2025, 41(7). PMID: 40581823; PMCID: PMC12240468. <a href="https://doi.org/10.1093/bioinformatics/btaf371">https://doi.org/10.1093/bioinformatics/btaf371</a></p>
				</ul>
			</div>
	</div>

	<div class="box">
	<div class="title_year">
		<p>2024</p>
	</div>
			<div class="Publications_text">
				<ul>
					<li><p><b><span style="color: black">Adversarial Robustness and Explainability of Machine Learning Models.</span></b><br />
					<span style="color:"><u>Gafur J,</u> <u>Goddard S,</u> <strong><u>Lai WKM.</u></strong> </span><br />
					<span style="color:black"><b>Practice and Experience in Advanced Research Computing</b></span> 2024, <a href="https://doi.org/10.1145/3626203.3670522">https://doi.org/10.1145/3626203.3670522</a> </p>

					<li><p><b><span style="color: black">Multi-dimensional analyses identify genes of high priority for pancreatic cancer research.</span></b><br />
					<span style="color:"><u>Nwosu ZC,</u> <u>Giza H,</u> <u>Nassif M,</u> <u>Charlestin V,</u> <u>Menjivar RE,</u> <u>Kim D,</u> <u>Kemp SB,</u> <strong><u>Lai WKM,</u></strong> <u>Loveless I,</u> <u>Steele NG,</u> <u>Hu J,</u> <u>Hu B,</u> <u>Wang S,</u> <u>Magliano MP,</u> <u>Lyssiotis CA.</u></span><br />
					<span style="color:black"><b>JCI Insight</b></span> 2024, 10(4):e174264. PMID: 39774001; PMCID: PMC11949049.<a href="https://doi.org/10.1172/jci.insight.174264">https://doi.org/10.1172/jci.insight.174264</a> </p>
				</ul>
			</div>
	</div>

	<div class="box">
	<div class="title_year">
		<p>2023</p>
	</div>
			<div class="Publications_text">
				<ul>
					<li><p><b><span style="color: black">GenoPipe: identifying the genotype of origin within (epi)genomic datasets.</span></b><br />
					<span style="color:"><u>Lang O,</u> <u>Srivastava D,</u> <u>Pugh BF,</u> <strong><u>Lai WKM.</u></strong> </span><br />
					<span style="color:black"><b>Nucleic Acids Research</b></span> 2023, 51 (22), 12054-12068. PMID: 37933851; PMCID: PMC10711449.</p>

					<li><p><b><span style="color: black">Joint sequence & chromatin neural networks characterize the differential abilities of Forkhead transcription factors to engage inaccessible chromatin.</span></b><br />
					<span style="color:"><u>Arora S,</u> <u>Yang J,</u> <u>Akiyama T,</u> <u>James DQ,</u> <u>Morrissey A,</u> <u>Blanda TR,</u> <u>Badjatia N,</u> <strong><u>Lai WKM,</u></strong> <u>Ko MSH,</u> <u>Pugh BF,</u> <u>Mahony S.</u></span><br />
					<span style="color:black"><b>bioArxiv</b></span> 2023, <a href="https://doi.org/10.1101/2023.10.06.561228">https://doi.org/10.1101/2023.10.06.561228</a> </p>
				</ul>
			</div>
	</div>

</div>

<footer id="footer" class="footer" role="contentinfo">
		<span>Copyright © 2025 William KM Lai</span>
</footer>
</div>
</div><!-- .overflow-container -->
