---
title: Software
---

<style>
.row1{text-align:center;}
.row2{text-align:center;}
.row3{text-align:center;}
.row4{text-align:center;}
.box1{ display:inline-block;
        width:40%;
        vertical-align: middle;
text-align: center;
margin:0 20px 0 20px
}
.box2{  display:inline-block;
        width:40%;
        vertical-align: middle;
        text-align: center;
        vertical-align: top;
margin:0 20px 0 20px
}
.box3{  display:inline-block;
        width:40%;
        vertical-align: middle;
        text-align: center;
margin:0 20px 0 20px
}
.box4{  display:inline-block;
        width:40%;
        vertical-align: middle;
        text-align: center;
vertical-align: top;
margin:0 20px 0 20px
}

</style>
<div class="Software">

  <h2 id="research-nn">Mission statement</h2>
    <div class="wp-block-media-text alignwide is-stacked-on-mobile is-vertically-aligned-top" style="grid-template-columns:20% auto">
      <div class="wp-block-media-text__content">
          <p class="has-normal-font-size">
          We develop and fund open-source 'big data' enabled AI analysis that is accessible for non-computational adept researchers working in biology labs. Our group is fully committed to FAIR compliance in all of our software development. We engage with the national community to develop, support, and disseminate novel technologies of benefit to all.
          </p>
  		</div>
  </div>

  <hr />
    <h4 id="research-nn">Infrastructure development</h4>
      <div class="wp-block-media-text alignwide is-stacked-on-mobile is-vertically-aligned-top" style="grid-template-columns:20% auto">
        <div class="wp-block-media-text__content">
        <p class="has-normal-font-size">Current genomic projects generate hundreds of terabytes of genomic sequencing data and associated metadata. Properly tracking, analyzing, and visualizing all of that data requires extensive management. We developed the STENCIL platform to curate a variety of Galaxy workflow into an integrated space while providing interactive analysis tools for further analysis. This platform was crucial to understanding and interpreting the data generated and provided a mechanism for biochemists with minimal genomic training to understand the results of the data generated. Additionally, in response to the NIHâ€™s continued support for enhanced experimental rigor and reproducibility, we also use and develop the PEGR platform to track the metadata associated with each experiment as it was performed. PEGR tracks all aspects of an experiment (i.e., enzyme catalog #, user ID) in real-time over the course of an experiment and embeds that information into a searchable interface that links directly to the downstream sequencing results and the Galaxy platform for linking reproducible bioinformatics all the way back to the original experimental design.</p>

        <h4>Highlights</h4>
        <ul>
          <li><p><b><span style="color: black">Protein structure prediction and design for high-throughput computing.</span></b><br />
          <span style="color:"><u>Mathew VS,</u> <u>Kellogg GD,</u> <strong><u>Lai WKM.</u></strong> </span><br />
          <span style="color:black"><b>bioArxiv</b></span> 2025, <a href="https://doi.org/10.1101/2025.07.18.665594">https://doi.org/10.1101/2025.07.18.665594</a> </p>

          <li><p><b><span style="color: black">Adversarial Robustness and Explainability of Machine Learning Models.</span></b><br />
          <span style="color:"><u>Gafur J,</u> <u>Goddard S,</u> <strong><u>Lai WKM.</u></strong> </span><br />
          <span style="color:black"><b>Practice and Experience in Advanced Research Computing</b></span> 2024, <a href="https://doi.org/10.1145/3626203.3670522">https://doi.org/10.1145/3626203.3670522</a> </p>

          <li><p><b><span style="color: black">GenoPipe: identifying the genotype of origin within (epi)genomic datasets.</span></b><br />
          <span style="color:"><u>Lang O,</u> <u>Srivastava D,</u> <u>Pugh BF,</u> <strong><u>Lai WKM.</u></strong> </span><br />
          <span style="color:black"><b>Nucleic Acids Research</b></span> 2023, 51 (22), 12054-12068. PMID: 37933851; PMCID: PMC10711449.</p>

          <li><p><b><span style="color: black">ScriptManager: an interactive platform for reducing barriers to genomics analysis.</span></b><br />
          <span style="color:"><u>Lang O,</u> <u>Pugh BF,</u> <u>Lai WKM</u> <strong><u>Lai WKM.</u></strong></span><br />
          <span style="color:black"><b>Practice and Experience in Advanced Research Computing</b></span> 2022, <a href="https://dl.acm.org/doi/fullHtml/10.1145/3491418.3535161">https://doi.org/10.1145/3491418.3535161</a> </p>

          <li><p><b><span style="color: black">PEGR: a flexible management platform for reproducible epigenomic and genomic research.</span></b><br />
          <span style="color:"><u>Shao D,</u> <u>Kellogg G,</u> <u>Nematbakhsh A,</u> <u>Kuntala PK,</u> <u>Mahony S,</u> <u>Pugh BF,</u> <strong><u>Lai WKM.</u></strong></span><br />
          <span style="color:black"><b>Genome Biology</b></span> 2022, 19;23(1):99. PMID: 35440038 PMCID: PMC9016988</p>

          <li><p><b><span style="color: black">STENCIL: A web templating engine for visualizing and sharing life science datasets.</span></b><br />
          <span style="color:"><u>Sun Q,</u> <u>Nematbakhsh A,</u> <u>Kuntala PK,</u> <u>Kellogg G,</u> <u>Pugh BF,</u> <strong><u>Lai WKM.</u></strong></span><br />
          <span style="color:black"><b>PLoS Computational Biology</b></span> 2022, 9;18(2):e1009859. doi: 10.1371/journal.pcbi.1009859. PMID: 35139076; PMCID: PMC8863220.</p>

          <li><p><b><span style="color: black">ArchTEx: accurate extraction and visualization of next-generation sequence data.</span></b><br />
          <span style="color:"><strong><u>Lai WKM,</u></strong> <u>Bard JE,</u> <u>Buck MJ.</u> </span><br />
          <span style="color:black"><b>Bioinformatics</b></span> 2012, 1;28(7):1021-3. doi: 10.1093/bioinformatics/bts063. PMID: 22302569.</p></li>
        </ul>

        <p><a href="/publications" data-type="page" data-id="5">See all publications</a></p>
        </div>
      </div>

<div class="row1">
	<div class="box1">

		<p><span style="font-weight: bold;"><strong><a href="https://github.com/CEGRcode/GenoPipe">GenoPipe</a></strong></span></p>
		<p><span style="font-size: 12px;">Quality control toolkit for characterizing the genotype of NGS datasets.</span></p>

	</div>
  <div class="box1">

		<p><span style="font-weight: bold;"><strong><a href="https://github.com/CEGRcode/scriptmanager">ScriptManager</a></strong></span></p>
		<p><span style="font-size: 12px;">GUI toolkit for NGS analysis scripts.</span></p>

	</div>
</div>



<div class="row2">
  <div class="box2">

    <p><span style="font-weight: bold;"><a href="https://github.com/seqcode/pegr">PEGR</a></strong></span></p>
    <p><span style="font-size: 12px;">Web service platform for biochemical metadata logging.</span></p>

  </div>
	<div class="box2">

		<p><span style="font-weight: bold;"><a href="https://github.com/CEGRcode/stencil">STENCIL</a></strong></span></p>
		<p><span style="font-size: 12px;">A web engine for visualizing and sharing life science datasets.</span></p>

	</div>
</div>

<div class="row3">
  <div class="box3">

    <p><span style="font-weight: bold;"><a href="https://github.com/WilliamKMLai/ArchAlign">ArchAlign</a></strong></span></p>
    <p><span style="font-size: 12px;">Coordinate-free chromatin alignment for epigenomic architecture identification.</span></p>

  </div>
	<div class="box3">

		<p><span style="font-weight: bold;"><strong><a href="https://github.com/WilliamKMLai/ArchTEx">ArchTEx</a></strong></span></p>
		<p><span style="font-size: 12px;">Accurate Extraction and Visualization of Next-Generation Sequence data.</span></p>

	</div>
</div>

</div>
